lib/SILOptimizer/IPO/CrossModuleOptimization.cpp

```diff
@@ -333,6 +401,18 @@ bool CrossModuleOptimization::canSerializeInstruction(SILInstruction *inst,
   if (auto *MI = dyn_cast<MethodInst>(inst)) {
     return !MI->getMember().isForeign;
   }
+  if (auto *SEAI = dyn_cast<StructElementAddrInst>(inst)) {
```

Comment:
This and the struct_extract case should be handled in `canSerializeType` instead. The code at the beginning of `canSerializeInstruction` will check whether we can serialize any type  involved in an operand or result value:

```  
 for (SILValue result : inst->getResults()) {                                  
    if (!canSerializeType(result->getType()))                                   
      return false;                                                             
  }                                                                             
  for (Operand &op : inst->getAllOperands()) {                                  
    if (!canSerializeType(op.get()->getType()))                                 
      return false;                                                             
  }
  ```
  
  If you add a check to canSerializeType that when we are compiling for resilience and serializing for package CMO the type must be loadable you catch any potential issue: including tuple(), tuple_extract, and any other cases I might have forgotten to list.

Reply:
To check whether a type is loadable in the minmal resilience domain (i.e once we marked the function as `[serialized]`) you can do something like:
```
auto inst = // is the instruction inside the function we are currently checking the type of
auto funcExp = inst->getFunction()->getTypeExpansionContext();
auto typeExpansion = TypeExpansionContext(ResilienceExpansion::Minimal, funcExp->getContext(), funcExp->isWholeModuleContext();
(SILModule)m.Types.getTypeLowering(silType, typeExpansion).isLoadable()
```